edb/server/pgcon/pgcon.pyx

```diff
@@ -947,10 +956,90 @@ cdef class PGProto:
         if er:
             raise pgerror.BackendError(fields=er)
 
-    async def restore(self, sql, bytes data):
+    cdef _elide_copy_cols(
+        self,
+        WriteBuffer wbuf,
+        char* data,
+        ssize_t data_len,
+        ssize_t ncols,
+        tuple elided_cols,
+    ):
+        """Rewrite the binary COPY stream to exclude the specified columns."""
+        cdef:
+            FRBuffer rbuf
+            ssize_t i
+            ssize_t real_ncols
+            int8_t *elide
+            int8_t elided
+            int32_t datum_len
+            char copy_msg_byte
+            int16_t copy_msg_ncols
+            const char *datum
+            bint first = True
+            bint received_eof = False
+
+        real_ncols = ncols + len(elided_cols)
+        frb_init(&rbuf, data, data_len)
+
+        elide = <int8_t*>cpythonx.PyMem_Calloc(
+            <size_t>real_ncols, sizeof(int8_t))
+
+        try:
+            for col in elided_cols:
+                elide[col] = 1
```

Comment:
This calloc and the initialization of `elide` can me be moved outside of this function to `async def _restore`